### PR TITLE
Memory editor

### PR DESCRIPTION
**Pull request by dpethes**
Port of ocornut's Mini memory editor for ImGui - https://gist.github.com/ocornut/0673e37e54aff644298b
Notes:
* needs a workaround code for ImGuiListClipper - it is not exposed through cimgui yet
* same for GetKeyIndex
* separator line drawing replaced by printing a pipe char to avoid custom rendering

### CODE REVIEW

File: src/ImGui.NET.SampleProgram/MemoryEditor.cs [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+                        if (AllowEdits && ImGui.IsLastItemHovered() && ImGui.IsMouseClicked(0))
+                        {
+                            DataEditingTakeFocus = true;
+                            DataEditingAddr = addr;
+                        }
+                    }
+                }
+
+                ImGui.SameLine(line_start_x + cell_width * Rows + glyph_width * 2);
+                //separator line drawing replaced by printing a pipe char
```
Comment (mellinoe) [inline src/ImGui.NET.SampleProgram/MemoryEditor.cs]: Which part of the gist doesn't work?

Comment (mellinoe) [inline src/ImGui.NET.SampleProgram/MemoryEditor.cs]: I guess it has something to do with the fact that none of the `DrawList` methods are wrapped 😄

**Comment by mellinoe** [comment]
This is very cool, thanks! I wonder if we could get the original line-drawing part from the gist to work if we added wrappers for the DrawList stuff?

File: src/ImGui.NET.SampleProgram/MemoryEditor.cs (hunk 1 above)
Comment (dpethes) [inline src/ImGui.NET.SampleProgram/MemoryEditor.cs]: I didn't want to get too low-level right from the start :D As a bonus, it now looks more like TotalCmd's hex viewer.

**Comment by dpethes** [comment]
I think it would work. Also, GetKeyIndex could be removed if it's added on top of version 1.49.

**Comment by mellinoe** [comment]
Thanks again, I'm going to merge this now.

**Pull request merged**